app/services/clients/base.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Clients for interacting with external APIs
+module Clients
+  # Base class for harvesting clients
+  class Base
+    attr_reader :conn
+
+    def initialize(conn: nil)
+      @conn = conn || new_conn
+    end
+
+    def get_json(path:, params: {})
+      response = conn.get(path, params.compact)
+
+      raise Clients::Error, "Error: #{response.status}" unless response.success?
+
+      JSON.parse(response.body)
+    rescue Faraday::Error => e
+      raise Error, "Connection err: #{e.message}"
+    rescue JSON::ParserError => e
+      raise Error, "JSON parsing error: #{e.message}"
+    end
+
+    private
+
+    def new_conn
+      Faraday.new('example.com')
+    end
+  end
+
+  # Result of a list operation
+  ListResult = Struct.new('ListResult', :id, :modified_token, keyword_init: true)
+
+  # Error class for client errors
+  class Error < StandardError; end
```

Comment:
At least in infra, we put the error classes and struct classes at the top of the surrounding class.

Reply:
Makes sense. Moved it to the top